diff --git a/node_modules/@viro-community/react-viro/components/ViroSurface.tsx b/node_modules/@viro-community/react-viro/components/ViroSurface.tsx
index eb69913..0512a30 100644
--- a/node_modules/@viro-community/react-viro/components/ViroSurface.tsx
+++ b/node_modules/@viro-community/react-viro/components/ViroSurface.tsx
@@ -96,7 +96,7 @@ export class ViroSurface extends ViroBase<Props> {
 }
 
 var VRTSurface = requireNativeComponent(
-  "VRTQuad",
+  "VRTSurface",
   // @ts-ignore
   ViroSurface,
   {
diff --git a/node_modules/@viro-community/react-viro/dist/components/ViroSurface.js b/node_modules/@viro-community/react-viro/dist/components/ViroSurface.js
index d381b27..da70e23 100644
--- a/node_modules/@viro-community/react-viro/dist/components/ViroSurface.js
+++ b/node_modules/@viro-community/react-viro/dist/components/ViroSurface.js
@@ -96,7 +96,7 @@ class ViroSurface extends ViroBase_1.ViroBase {
     }
 }
 exports.ViroSurface = ViroSurface;
-var VRTSurface = (0, react_native_1.requireNativeComponent)("VRTQuad", 
+var VRTSurface = (0, react_native_1.requireNativeComponent)("VRTSurface", 
 // @ts-ignore
 ViroSurface, {
     nativeOnly: {
